### DISCUSSION TITLE
Slides with Task List

### DISCUSSION THREAD

**Discussion by damian-krawczyk** [marp-team/marp#352] [Ideas] [answered]
Is it possible to render in Marpit slide Markdown [Task Lists](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-task-lists#creating-task-lists)? I think it's not possible (at least with default setup).

Would be great if we could quickly present todo items with rendered checkboxes. Can you add this functionality?

---
# Project list to do

- [ ] Item 1
    - [x] Subitem 1.1
    - [ ] Subitem 1.2
- [x] Item 2

**Comment by yhatt** [comment, marked as answer]
The task list syntax is out of CommonMark spec so [we leave it to the plugin](https://marpit.marp.app/usage?id=extend-marpit-by-plugins#:~:text=Due%20to%20our%20policy%2C%20Marpit%20has%20not%20extended%20Markdown%20syntax%20such%20as%20to%20break%20CommonMark.%20But%20you%20may%20use%20plugins%20if%20you%20want%20the%20aggressive%20extended%20syntax.).

[You can extend Marpit rendering engine](https://marpit.marp.app/usage?id=extend-marpit-by-plugins) with [`markdown-it-task-lists`](https://github.com/revin/markdown-it-task-lists) markdown-it plugin.

```javascript
const { Marpit } = require('@marp-team/marpit');
const taskLists = require('markdown-it-task-lists');

const marpit = new Marpit().use(taskLists);
const result = marpit.render('...'); // markdown string containing task list items
```

Example in Marp CLI:

```javascript
// engine.js
const taskLists = require('markdown-it-task-lists');
module.exports = ({ marp }) => marp.use(taskLists);
```

```bash
npm i --save-dev @marp-team/marp-cli markdown-it-task-lists
npx marp --engine ./engine.js ./md-with-task-lists.md
```

- Related
  - https://github.com/marp-team/marp-core/issues/238
